differ/differ.go

```diff
@@ -114,27 +113,22 @@ func NewFileAttribute(fileName string) *FileAttributes {
 
 type oneEntry struct {
 	Key               string
+	CrMeta            *crMeta.CRMetadata
```

Comment:
Given 
1. `String()` method could be called at anytime
2. `CrMeta` is supposed to be there no matter what the condition is
It may be safer to use `crMeeta.CRMetadata` as a reference instead of a pointer

Other function calls will need to adjust (i.e. pass in `&crMeta.CRMetadata` instead)

Reply:
Is it ok if i add a check in the String() function(crMeta==nil)?.
I am trying to analyse as to why we should do that.